### PR TITLE
feat(quantic): script community deployment

### PR DESCRIPTION
**Pull request by lbergeron**
https://coveord.atlassian.net/browse/SFINT-4044

This PR makes it easier to setup the examples community by adding scripts to the Quantic package.

3 scripts are added:

* `setup:examples` which takes care of creating, configuring, and deploying the community from scratch. It reuses the scratch org and community if they already exist.
* `deploy:main` which deploys only the `main` package to the `LWC` org.
* `deploy:examples` which deploys only the `examples` package to the `LWC` org.

All scripts rely on `sfdx-cli` installed as a dev dependency. One reason is to make sure to avoid the bug where `sfdx-cli` deletes the unit tests when deploying packages.

Here is what the output of `npm run setup:examples` looks like:

    [1/6] Searching for LWC organization...
    [1/6] LWC organization found.
    [2/6] Searching for 'Quantic Examples' community
    [2/6] Community found.
    [3/6] Deploying components...
    [3/6] Components deployed.
    [4/6] Deploying community metadata...
    [4/6] Community metadata deployed.
    [5/6] Publishing community...
    [5/6] Community published.
    [6/6] Searching for configuration file 'path-to-your-working-copy/packages/quantic/cypress/plugins/config/examples-community.json'...
    [6/6] Creating configuration file...
    [6/6] Configuration file updated.

    The 'Quantic Examples' community is ready, you can access it at the following URL:
    https://your-salesforce-scratch-org-instance.force.com/examples

    To open Cypress, run:
    npm run cypress:open

### CODE REVIEW

File: packages/quantic/scripts/build/deploy-community.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    (o) => o.alias === options.alias
+  );
+
+  if (!!org && org.status !== 'Active') {
+    console.warn(
+      `Org ${options.alias} is found but status is not active. Status is ${org.status}.`
+    );
+  }
+
+  return !!org && org.status === 'Active';
```
Comment (jeremierobert-coveo) [inline packages/quantic/scripts/build/deploy-community.ts:38]:
```suggestion
  const isOrgActive = !!org && org.status !== 'Active';

  if (isOrgActive) {
    console.warn(
      `Org ${options.alias} is found but status is not active. Status is ${org.status}.`
    );
  }

  return isOrgActive;
```

**Review by jeremierobert-coveo — APPROVED** [review]
Very nice setup 😃 !
(Would you mind at some point bringing this setup in the sfint package ?)

**Review by nathanlb — APPROVED** [review]
T'es une vrai machine @lbergeron!
C'est :vraimentvraimentnice:

**Comment by lbergeron** [comment]
Thanks @jeremierobert-coveo @nathanlb ! I'm quite happy with the result.

And yes, I guess we should be able to bring parts of this setup to improve sfint.

File: packages/quantic/README.md [hunk 2]
```diff
@@ -54,43 +54,29 @@ Start the server.
...
 
-    ```bash
-    sfdx force:community:publish -u LWC -n "Quantic Examples" --json
-    ```
+To open Cypress, run:
+npm run cypress:open
+```
 
-    The community URL will appear in the console output.
+Once the community has been deployed, you can only deploy the `main` or `example` components when needed. To do so, run:
```
[the remark this answers is not in the document]
Comment (jpmarceau) [reply packages/quantic/README.md:73]: (Ici, je suppose que tu veux dire que, une fois la communauté déployée, il est possible de déployer seulement `main` ou seulement `example`, si tu veux. Dans ce cas, tu pourrais tout déployer ensemble aussi.

Ce que tu as d'écrit présentement voudrait plutôt dire que, une fois la communauté déployée, il serait uniquement possible de déployer `main` ou `example` un à la fois, quand tu en as besoin. Il ne serait plus possible de déployer tout en même temps. Si c'était ce que tu veux dire je suggérerais autre chose)

File: packages/quantic/scripts/build/deploy-community.ts (hunk 1 above)
Comment (lbergeron) [reply packages/quantic/scripts/build/deploy-community.ts:38]: I went with something slightly different than you suggested. Let me know what you think of it.

    const isOrgFound = !!org;
    const isOrgActive = isOrgFound && org.status === 'Active';

    if (isOrgFound && !isOrgActive) {
        console.warn(
            `Org ${options.alias} is found but status is not active. Status is ${org.status}.`
        );
    }

    return isOrgActive;

File: packages/quantic/README.md [hunk 3]
```diff
@@ -54,43 +54,29 @@ Start the server.
...
+```
 
-    ```bash
-    sfdx force:mdapi:deploy -u LWC -d ./quantic-examples-community -w 5
-    ```
+This script creates, configures, and deploys everything required to have fully working examples. The community URL is provided at the end of the script like this.
 
-4. To publish the community, run:
+```plain
+The 'Quantic Examples' community is ready, you can access it at the following URL:
```
[the remark this answers is not in the document]
Comment (lbergeron) [reply packages/quantic/README.md:66]: I had the same thought, but I wasn't sure enough :smile: . I'll go with your suggestion.

File: packages/quantic/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lbergeron) [reply packages/quantic/README.md:73]: > je suppose que tu veux dire que, une fois la communauté déployée, il est possible de déployer seulement main ou seulement example, si tu veux.

C'est exactement ce que je voulais dire. `deploy:main` et `deploy:examples` prennent beaucoup moins de temps à exécuter que `setup:examples` qui refait le setup au complet.

**Review by jpmarceau — APPROVED** [review]
👌

File: packages/quantic/scripts/build/deploy-community.ts (hunk 1 above)
Comment (jeremierobert-coveo) [reply packages/quantic/scripts/build/deploy-community.ts]: that's perfect

**Pull request merged**